granularflow.go

```diff
@@ -192,7 +192,7 @@ func (c *protocolConnection) prepare(r *buff.Reader, q *gfQuery) error {
 	for r.Next(done.Chan) {
 		switch r.MsgType {
 		case message.PrepareComplete:
-			ignoreHeaders(r)
+			c.cacheHeaders(q, decodeHeaders(r))
```

Comment:
I would be granular here and cache the returned capabilities word explicitly.  Caching all headers indiscriminately is likely incorrect as you might accidentally rely on a piece of data that isn't actually stable.

Reply:
changed